docs/design/persistent-storage.md

```diff
@@ -0,0 +1,220 @@
+
+A `PersistentStorageController` manages the creation, attachment, and reclamation of PersistentStorage objects for one VolumeSource.  Implementations must be pluggable to support many types of controllers for many types of VolumeSources. The controller can advise the scheduler on host selection, keeping volume rules contained within the provider.  All cloud provider credentials are isolated to the storage controller's host.
+
+`PersistentStorageBinder` watches for pods with persistence requirements and matches the request with what is available in the pool.
+
+`PersistenceRequirements` expresses the capabilities and performance attributes of the underlying storage.  Allow discovery
+of storage by its capabilities and use those capabilities as pod storage requirements.
+
+Kubernetes makes no guarantees at runtime that the underlying storage exists or is available.  High availability is left to the storage provider.
+
+## Goals
+
+* Allow administrators to describe available storage
+* Allow pod authors to discover and request persistence for their volumes
+* Enforce security through access control lists and securing storage to the same namespace as the pod volume
+* Enforce quotas through admission control
+* Ensure developers can rely on storage being available without being closely bound to a particular disk, server, network, or storage device.
+
+
+#### Allow admins to describe and create available storage
+
+Admins can describe storage with `PersistentStorageControllers` or create storage directly by using the `PersistentStorage` API.  Storage created manually still requires a PersistentStorageController and accompanying interface to allow management by the cluster.
+
+
+```
+	
+	PersistentStorage
+		TypeMeta
+		ObjectMeta
+			Name - internal name
+			Labels - labeled like Volume.PersistenceRequirements.Selector
+		Spec
+			VolumeSource
+				HostPath
+				EmptyDir
+				GitRepo	
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany }
+				AWSElasticBlockStore { modes: ReadWriteOnce }
+				NFSMount { modes: ReadWriteOnce | ReadOnlyMany | ReadWriteMany }
+				*** many volume plugins
+			RetentionPolicy	 -- volumes can be orphaned. allow never delete.
+		Status
+			Phase
+			CurrentMounts	MountList
+			LastMounts		MountList
+			PodRef			ObjectReference -- the pod originally requesting persistence
+			
+
+	// Manages one type of PersistentStorage
+	PersistentStorageController
+		Name
+		Spec
+			VolumeSource
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany, resources: 5gb 30iops, .48 throughput  }
```

Comment:
how are you going to materialize resources? there are many vendor specific capabilities on QoS and HA and they don't always use the same terms. Some can be found from Cinder Volume Type.